server/plugin.go

```diff
@@ -153,6 +153,12 @@ func (p *Plugin) handleTelemetry(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if telemetryRequest == nil {
```

Comment:
I think a better solution would be to have the code above this block changed to:

```diff
var telemetryRequest *telemetryAPIRequest
decoder := json.NewDecoder(r.Body)
- err := decoder.Decode(&telemetryRequest)
+ err := decoder.Decode(telemetryRequest)
```

This way the decoder can return an error when `null` is passed in through the API. See the exmaple here https://goplay.tools/snippet/uw72B-fWZFa

Reply:
@mickmister I have thought of a similar solution. The only thing was the error decoder is returning doesnt look that good. Still have made that change.
```
{"error":"Unable to decode JSON","details":"json: Unmarshal(nil *main.addAPIRequest)"}
```

Reply:
@mickmister I tried the above code. But i dont think this is working and the decoder is giving the error in every case not just for the null case. Please see the example below
https://goplay.tools/snippet/5BFDGIxdNPs